app/scripts/metamask-controller.js

```diff
@@ -1135,17 +1136,30 @@ export default class MetamaskController extends EventEmitter {
     });
 
     ///: BEGIN:ONLY_INCLUDE_IF(snaps)
+    const shouldUseOffscreenExecutionService =
+      isManifestV3 &&
+      typeof chrome !== 'undefined' &&
+      // eslint-disable-next-line no-undef
+      typeof chrome.offscreen !== 'undefined';
+
     const snapExecutionServiceArgs = {
-      iframeUrl: new URL(process.env.IFRAME_EXECUTION_ENVIRONMENT_URL),
+      [shouldUseOffscreenExecutionService ? 'frameUrl' : 'iframeUrl']: new URL(
```

Comment:
`frameUrl` is no longer a constructor argument for `OffscreenExecutionService`. Instead the URL is hardcoded and tied to the version of the executor package.

Reply:
@FrederikBolding so i need to just conditionally add the 'iframeUrl' parameter for the IframeExecutionService?

Reply:
What you did now looks fine to me! 👍